tests/unit/test_run.py

```diff
@@ -0,0 +1,1788 @@
+        res.filter(variable="Radiative Forcing", year=1992, month=1, day=1).values,
+        np.average(np.arange(6)),
+    )
+    npt.assert_allclose(
+        res.filter(variable="Radiative Forcing", year=1996, month=1, day=1).values,
+        np.average([42, 43, 44]),
+    )
+
+    # automate rest of tests
+    def group_annual_mean_beginning_of_year(x):
+        if x.month <= 6:
+            return x.year
+        return x.year + 1
+
+    ts_resampled = (
+        test_scm_df_monthly.timeseries()
+        .T.groupby(group_annual_mean_beginning_of_year)
+        .mean()
+        .T
+    )
+    ts_resampled.columns = ts_resampled.columns.map(lambda x: dt.datetime(x, 1, 1))
+
+    pd.testing.assert_frame_equal(res.timeseries(), ts_resampled, check_like=True)
+
+
+def test_time_mean_year(test_scm_df_monthly):
+    # should be annual mean (using all values in that year)
+    res = test_scm_df_monthly.time_mean("AC")
+
+    # test by hand
+    npt.assert_allclose(
+        res.filter(variable="Radiative Forcing", year=1992, month=7, day=1).values,
+        np.average(np.arange(12)),
+    )
+    npt.assert_allclose(
+        res.filter(variable="Radiative Forcing", year=1995, month=7, day=1).values,
+        np.average(np.arange(36, 45)),
+    )
+
+    # automate rest of tests
+    def group_annual_mean(x):
+        return x.year
+
+    ts_resampled = (
+        test_scm_df_monthly.timeseries().T.groupby(group_annual_mean).mean().T
+    )
+    ts_resampled.columns = ts_resampled.columns.map(lambda x: dt.datetime(x, 7, 1))
+
+    pd.testing.assert_frame_equal(res.timeseries(), ts_resampled, check_like=True)
+
+
+def test_time_mean_year_end_of_year(test_scm_df_monthly):
+    # should be annual mean centred on December 31st of each year
+    res = test_scm_df_monthly.time_mean("A")
+
+    # test by hand
+    npt.assert_allclose(
+        res.filter(variable="Radiative Forcing", year=1991, month=12, day=31).values,
+        np.average(np.arange(6)),
+    )
+    npt.assert_allclose(
+        res.filter(variable="Radiative Forcing", year=1995, month=12, day=31).values,
+        np.average(np.arange(42, 45)),
+    )
+
+    # automate rest of tests
+    def group_annual_mean_end_of_year(x):
+        if x.month >= 7:
+            return x.year
+        return x.year - 1
+
+    ts_resampled = (
+        test_scm_df_monthly.timeseries()
+        .T.groupby(group_annual_mean_end_of_year)
+        .mean()
+        .T
+    )
+    ts_resampled.columns = ts_resampled.columns.map(lambda x: dt.datetime(x, 12, 31))
+
+    pd.testing.assert_frame_equal(res.timeseries(), ts_resampled, check_like=True)
+
+
+def test_time_mean_unsupported_style(test_scm_df_monthly):
+    error_msg = re.escape("`rule` = `junk` is not supported")
+    with pytest.raises(ValueError, match=error_msg):
+        test_scm_df_monthly.time_mean("junk")
+
+
+def test_set_meta_as_series(test_scm_run):
+    s = pd.Series([0.3, 0.4, 0.5])
+    test_scm_run.set_meta(s, "meta_series")
```

Comment:
is it worth getting rid of setting from a series and swapping to it's `set_meta(key, value)` in this PR or should we do it in a follow up?

Reply:
Follow up. I tried to keep as few changes as possible

#40